pkg/agent/datamodel/types.go

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+package datamodel
+
+import (
+	"strings"
+
+	"github.com/Azure/aks-engine/pkg/api"
+	"github.com/Azure/aks-engine/pkg/helpers"
+)
+
+// ContainerService complies with the ARM model of
+// resource definition in a JSON template.
+type ContainerService struct {
+	ID       string                    `json:"id"`
+	Location string                    `json:"location"`
+	Name     string                    `json:"name"`
+	Plan     *api.ResourcePurchasePlan `json:"plan,omitempty"`
+	Tags     map[string]string         `json:"tags"`
+	Type     string                    `json:"type"`
+
+	Properties *api.Properties `json:"properties,omitempty"`
+}
+
+// GetCloudSpecConfig returns the Kubernetes container images URL configurations based on the deploy target environment.
+//for example: if the target is the public azure, then the default container image url should be k8s.gcr.io/...
+//if the target is azure china, then the default container image should be mirror.azure.cn:5000/google_container/...
+func (cs *ContainerService) GetCloudSpecConfig() api.AzureEnvironmentSpecConfig {
+	targetEnv := helpers.GetTargetEnv(cs.Location, cs.Properties.GetCustomCloudName())
+	return api.AzureCloudSpecEnvMap[targetEnv]
+}
+
+// IsAKSCustomCloud checks if it's in AKS custom cloud
+func (cs *ContainerService) IsAKSCustomCloud() bool {
+	return cs.Properties.CustomCloudEnv != nil &&
+		strings.EqualFold(cs.Properties.CustomCloudEnv.Name, "akscustom")
+}
+
+// FromAksEngineContainerService converts aks-engine's ContainerService to our
+// own ContainerService. This is temporarily needed while we are still using
+// aks-engine's ApiLoader to load ContainerService from file. Once that code
```

Comment:
a little bit context here - the ApiLoader function is not(will not be) in a prod code path of AKS, since AKS always pass in the full container service object for template generation. That's why we don't need many of the control plane side default settings in agentbaker.

Reply:
Hi, Tony,
Since ApiLoader is not used in prod code path, how is ApiLoader used in AgentBaker? Is it used in testing, or tooling, or something else?